[WIP] fix approve version layers response handling

camallen opened the pull request:
Approve version layers status reporting replies on the raw response objects, but the API functions return either parse JSON body or an error object.

camallen committed:
add thoughts on approval status reporting

status reporting replies on the raw response objects, but the API 
functions return either parse JSON body or an error object.

camallen commented on js/maps.js:
```diff
@@ -78,14 +78,31 @@ function buildLayerGroup(versionGroup) {
+          // }]
+
+          // I think this has to reflect on the response obj here to decide what todo
+          // its either the body and all good
```
I'm unsure how to switch the behaviour here, how does one reflect on the different return values?

eatyourgreens replied:
If you want to switch based on the response status, you'll need to do that in the API code: https://github.com/zooniverse/PRN-maps/blob/a64902e31db45028b36ca38e8abc5c25d926ef18/js/api.js#L14-L25

camallen replied:
I've updated the API post method to throw now so the calling code can catch.

camallen committed:
fix the pending submission button

accept the json response as ok status

modify API post error response to throw the error to the calling code 
after logging.
